openedx/core/djangoapps/programs/signals.py

```diff
@@ -140,68 +84,84 @@ class of the object instance that sent this signal
     from openedx.core.djangoapps.programs.tasks import revoke_program_certificates
     revoke_program_certificates.delay(user.username, str(course_key))
 
 
 @receiver(COURSE_CERT_DATE_CHANGE, dispatch_uid='course_certificate_date_change_handler')
-def handle_course_cert_date_change(sender, course_key, **kwargs):  # lint-amnesty, pylint: disable=unused-argument
+def handle_course_cert_date_change(sender, course_key, **kwargs):  # pylint: disable=unused-argument
     """
-    If a course-run's `certificate_available_date` is updated, schedule a celery task to update the `visible_date`
-    attribute of all (course) credentials awarded in the Credentials service.
+    When a course run's configuration has been updated, and the system has detected an update related to the display
+    behavior or availability date of the certificates issued in that course, we should enqueue celery tasks responsible
+    for:
+        - updating the `visible_date` attribute of any previously awarded certificates the Credentials IDA manages
+        - updating the certificate available date of the course run's course certificate configuration in Credentials
 
     Args:
-        course_key(CourseLocator): refers to the course whose certificate_available_date was updated.
+        course_key(CourseLocator): The course run key of the course run which was updated
+
+    Returns:
+        None
     """
-    # Import here instead of top of file since this module gets imported before the credentials app is loaded, resulting
-    # in a Django deprecation warning.
-    from openedx.core.djangoapps.credentials.models import CredentialsApiConfig
-
-    # Avoid scheduling new tasks if we're not using the Credentials IDA
-    if not CredentialsApiConfig.current().is_learner_issuance_enabled:
-        LOGGER.warning(
-            f"Skipping handling of COURSE_CERT_DATE_CHANGE for course {course_key}. Use of the Credentials service is "
-            "disabled."
-        )
+    if not is_credentials_enabled():
         return
 
     LOGGER.info(f"Handling COURSE_CERT_DATE_CHANGE for course {course_key}")
     # import here, because signal is registered at startup, but items in tasks are not yet loaded
-    from openedx.core.djangoapps.programs.tasks import update_certificate_visible_date_on_course_update
     from openedx.core.djangoapps.programs.tasks import update_certificate_available_date_on_course_update
-    # update the awarded credentials `visible_date` attribute in the Credentials service after a date update
+    from openedx.core.djangoapps.programs.tasks import update_certificate_visible_date_on_course_update
     update_certificate_visible_date_on_course_update.delay(str(course_key))
-    # update the (course) certificate configuration in the Credentials service after a date update
     update_certificate_available_date_on_course_update.delay(str(course_key))
+
+
+@receiver(COURSE_PACING_CHANGED, dispatch_uid="update_credentials_on_pacing_change")
```

Comment:
*New behavior*

This fixes an edge case where an instructor-paced course with a certificate available date is updated to self-paced. Previously, the monolith would update the course, but the certificate available date would stay in place in Credentials. This can cause issues where certificates are not visible to learners on their Program Records (in Credentials), but the certificates themselves are available in the LMS by the learners. 

This is a confusing state. We should try to clean up data downstream when possible.